### PR TITLE
[FIX] mail: notification not sent when using the full composer

### PR DESCRIPTION
**Pull request by pka-odoo**
Current behavior before PR:
mail notification not sent when using the full composer.

Desired behavior after PR is merged:
mail notification will send when using the full composer.

LINKS:
Task-2446855

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/crm/tests/test_performances.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def test_assign_perf_populated(self):
                 lead.probability = (idx + 1) * 10 * ((int(lead.priority) + 1) / 2)
 
         with self.with_user('user_sales_manager'):
-            with self.assertQueryCount(user_sales_manager=6250):  # crm only: 6237
+            with self.assertQueryCount(user_sales_manager=6290):  # crm only: 6237
```
Comment (seb-odoo) [inline addons/crm/tests/test_performances.py:177]: Please run the test in local with just `crm` installed and update the comment to new value as well (if it changed?).

**Comment by tde-banana-odoo** [comment]
Really not sure for that one. It may have an impact on some flows. As asked on original PR
"""
Hem ... obviously the fix is right. However, can't it be considered as a functional change ? Since which version is it broken ? Indeed there is a difference between logging without subtype (no one is notified) and logging with not subtype (people following notes are notified). Did you grep all message_post in code to check if there could be unwanted changes ?
"""

File: addons/crm/tests/test_performances.py (hunk 1 above)
Comment (pka-odoo) [reply addons/crm/tests/test_performances.py]: @seb-odoo Done :+1:

**Comment by seb-odoo** [comment]
@tde-banana-odoo so what you suggest is that we should delete the line (because it is useless anyway) and always pass `subtype_id`/`subtype_xmlid` for note when it is what we want?

How comes this question was not raised the multiple times this line was either moved or changed before, but arising now?
The original intent must have been to always fallback to note? Why is that put into question now?
Do we have to assume people since called `message_post` in a way that relied on this bug?

**Comment by tde-banana-odoo** [comment]
I don't suggest anything... I think this PR requires to go through all message_post / existing in standard code and see if we may change anything.  Maybe we will have to force some subtype_ids to False in some cases for example, or find some other strange cases. We cannot change a default parameter in such an heavily used method without reading all callers. This is part of the bugfix, even if it is boring to do :) .

**Comment by seb-odoo** [comment]
I completely forgot that I actually reported this bug on the commit last year:
https://github.com/odoo/odoo/commit/e2b33f460d0f2359e65107f8615c7e827350b4e7#r40085148

I don't want to check every single of 565 calls to `message_post` in pretty much all business modules to try to guess if their intent was to post a note or post no subtype.

Before the commit introducing the issue it was always note and nobody complained. Now it's not a note and people have issues.

I think this should be fixed in 14 as-is. If people somehow receive too many notifications from the change they can always unfollow notes from the threads that generate these notifications that they don't want.

**Comment by tde-banana-odoo** [comment]
Ok, will have a look asap. I will try to find message_post without subtypes just to have a global look. 565 calls is easy to do :D .

**Comment by seb-odoo** [comment]
This was done at https://github.com/odoo/odoo/pull/66421

**Pull request closed without merging**